cpuid.go

```diff
@@ -305,325 +292,332 @@ func Detect() {
 
 // Generated here: http://play.golang.org/p/BxFH2Gdc0G
 
+func inSet(set FlagSet, flag string) bool {
```

Comment:
> but it should be fast enough that it can be called fairly regularly and as map lookup isn't suited for that even if it is fairly fast

FWIW, there's also a `Map` from `sync` which could be an option here (optimized for caches with multiple reads).

Reply:
> FWIW, there's also a Map from sync which could be an option here (optimized for caches with multiple reads).

You can read concurrently from a `map` as well, so we don't need that additional feature. Plus it is even slower than an unsynchronized `map` and involves type casting. So I don't want to go down that route.